cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/Setup.java

```diff
@@ -15,85 +16,119 @@
 import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserKeyPair;
 import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserPublicKey;
 import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserSecretKey;
+import org.cryptimeleon.math.hash.impl.SHA256HashFunction;
 import org.cryptimeleon.math.structures.groups.GroupElement;
 import org.cryptimeleon.math.structures.groups.cartesian.GroupElementVector;
+import org.cryptimeleon.math.structures.groups.debug.DebugBilinearGroup;
 import org.cryptimeleon.math.structures.groups.elliptic.BilinearGroup;
 import org.cryptimeleon.math.structures.groups.elliptic.type3.bn.BarretoNaehrigBilinearGroup;
 import org.cryptimeleon.math.structures.rings.cartesian.RingElementVector;
 import org.cryptimeleon.math.structures.rings.zn.Zn;
 import org.cryptimeleon.math.structures.rings.zn.Zn.ZnElement;
 
+import java.math.BigInteger;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
 /**
  * a class implementing the incentive system algorithms (of the incentive system from the 2020 paper) that are not part of any protocol
  * (namely Setup, P.KeyGen, U.KeyGen)
  */
 public class Setup {
-    public static final int PRF_KEY_LENGTH = 128;
+    // Which AES version to use for the PRF-to-ZN
+    public static final int HASH_THEN_PRF_AES_KEY_LENGTH = 256;
+
+    // (1/2)^OVERSUBSCRIPTION is a lower bound for probability of the PRF failing
+    private static final int HASH_THEN_PRF_OVERSUBSCRIPTION = 128;
+
+    // Base for esk decomposition. Requires as many signed digits in the provider's public key, but reduces digits and
+    // hence verifications required in the Spend-Deduct protocol for the encryption secret key
+    private static final long ESK_DEC_BASE = 256;
+
+    // ESK_DEC_BASE^MAX_POINTS_BASE_POWER determines the maximum number of points a user can have.
+    // This is due to the CCS range proof used for v>=k in Spend-Deduct
+    private static final int MAX_POINTS_BASE_POWER = 6;
```

Comment:
Is there any reason for the choice of 256^6 as max point count?

Reply:
I think it should be sufficient but not too large (2^48 points should be out of scope, and not too large.) We could also reduce the size, but the speedup would be not huge considering the ~50-100 SetMembershipProofs for the esk digits

Reply:
We could consider making this a parameter of the setup method, or overload it and make 6 a reasonable default.